apps/aecore/lib/aecore/structures/data_tx.ex

```diff
@@ -72,6 +73,11 @@ defmodule Aecore.Structures.DataTx do
     accounts_state = chainstate.accounts
     tx_type_state = Map.get(chainstate, tx.type, %{})
 
+    if !nonce_valid?(accounts_state, tx) do
+      Logger.error("Nonce is too small")
```

Comment:
Looks good, just one thing - if we log before throwing here I think we will log the same thing 2 times since the caught message is logged in a different place in this PR https://github.com/aeternity/elixir-node/pull/319/files#diff-d36d6614178f722c7eba656b1410801fR97